src/tracer.js

```diff
@@ -228,6 +228,7 @@ export default class Tracer {
             ctx.parentId = null;
             ctx.flags = flags;
         } else {
+            this.processDeferredSampling(parent, operationName, internalTags);
```

Comment:
The problem with the approach you suggest is that we can no longer guarantee the consistency of the sampling decision for all children (because each child has an independent sampling decision). We need to store the sampling decision for all future spans in the parent span context object or something associated with it to ensure all child spans have the same sampling decision applied to them.

I don't think that this is a easy problem to solve neatly while most of our stack assumes that sampling decisions happen when the span context is initially created.